### PR TITLE
Fixes #106 Ruchy cannot find by waypoint

### LINKED ISSUE #106: [Ruchy] Cannot find by waypoint
Related to PR #77

Search by waypoint is now failing.

**Step to reproduce**
1. Open page https://boly38.gk.kumy.org/ruchy.php
1. Fill the "Waypoint" field with any known waypoint such as `GC5BRQK`

**Actual result**
* The lookup via https://boly38.gk.kumy.org/szukaj-ajax.php?skad=ajax&wpt=GC5BRQK report `{"tresc":"\"error\" No cache found"}`. A red cross is shown on the right.
* The `coordinates` field is filled with `undefined undefined`.

**Expected result**
* It should display some details information details about the cache on the right (green mark and country flag).
* The `coordinates` field should be filled with coordinates.

If there is no match at all, the red mark has to be displayed on the right, and a message indicating that "user has to fill coordinates manually" should be displayed.

### PR DESCRIPTION
**Pull request by boly38**
fix default lat/lon to empty
fix cache without lat/lon when only one result
re add legacy waypoint search via gk-ruchy table
fix link label when no cache name
add flag image on ruchy based result
display waypoint uppercase
handle case where piwik is disabled
improve exception handler

Co-authored-by: Mathieu Alorent <github@kumy.net>

### CODE REVIEW

**Comment by kumy** [comment]
Please split your commits by features. I expect one commit for each of the points you listed in the commit description. As is, it s really difficult to review

On June 22, 2018 1:03:48 PM GMT+02:00, Boly38 <notifications@github.com> wrote:
>fix default lat/lon to empty
>fix cache without lat/lon when only one result
>re add legacy waypoint search via gk-ruchy table
>fix link label when no cache name
>add flag image on ruchy based result
>display waypoint uppercase
>handle case where piwik is disabled
>improve exception handler
>
>Co-authored-by: Mathieu Alorent <github@kumy.net>
>You can view, comment on, or merge this pull request online at:
>
>  https://github.com/geokrety/geokrety-website/pull/113
>
>-- Commit Summary --
>
>  * Fixes #106 Ruchy cannot find by waypoint
>
>-- File Changes --
>
>    M website/lib/Waypointy.class.php (76)
>    M website/ruchy.php (51)
>    M website/szukaj-ajax.php (44)
>
>-- Patch Links --
>
>https://github.com/geokrety/geokrety-website/pull/113.patch
>https://github.com/geokrety/geokrety-website/pull/113.diff
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/geokrety/geokrety-website/pull/113

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by boly38** [comment]
oops, following squash I loose git history.
anyway, I will try to divide the commit in 2/3

File: website/ruchy.php [hunk 1]
```diff
@@ -477,7 +496,8 @@
         //include_once("timeline.php");
     } // for each numer in multilog
 
-    if ($kret_multilog == '1') {
+    if (isset($errors)) {
+    } elseif ($kret_multilog == '1') {
```
Comment (kumy) [inline website/ruchy.php]:
```
+    if (isset($errors)) {
+    } elseif ($kret_multilog == '1') {
```
This pattern seems quite unusual for me, may you eventually add a comment or write it differently?

File: website/lib/Waypointy.class.php [hunk 2]
```diff
@@ -6,37 +6,105 @@ class Waypointy {
...
+        }
+        if (!$stmt->bind_param('s', $waypoint)) {
+            throw new Exception($action.' binding parameters failed: ('.$stmt->errno.') '.$stmt->error);
+        }
+        if (!$stmt->execute()) {
+            throw new Exception($action.' execute failed: ('.$stmt->errno.') '.$stmt->error);
+        }
+        $this->waypoint = $waypoint;
+        if ($this->isGCWaypoint($waypoint)) {
+            $this->cache_link = "http://www.geocaching.com/seek/cache_details.aspx?wp=$waypoint";
```
Comment (kumy) [inline website/lib/Waypointy.class.php]: http**s** it will prevent one redirect ;)

File: website/lib/Waypointy.class.php [hunk 3]
```diff
@@ -6,37 +6,105 @@ class Waypointy {
...
+    public function isImportedWaypoint($waypoint) {
+        return $this->isOCWaypoint($waypoint) or $this->isWPGWaypoint($waypoint);
+    }
+
+    public function getByWaypointFromRuchy($waypoint) {
+        $action = 'Waypointy::getByWaypointFromRuchy';
+
+        if (!($stmt = $this->dblink->prepare(self::SELECT_RUCHY
+                       .' WHERE `waypoint` LIKE ? ORDER BY `data_dodania` DESC LIMIT 1'))) {
+            throw new Exception($action.' prepare failed: ('.$this->dblink->errno.') '.$this->dblink->error);
```
Comment (kumy) [inline website/lib/Waypointy.class.php]: :heart_eyes:
